src/commands/dependencies/upload.ts

```diff
@@ -0,0 +1,157 @@
+import {Command} from 'clipanion'
+import {BufferedMetricsLogger} from 'datadog-metrics'
+import fs from 'fs'
+import path from 'path'
+
+import {apiConstructor} from './api'
+import {Payload} from './interfaces'
+import {getMetricsLogger} from './metrics'
+import {
+  renderCannotFindFile,
+  renderCommandInfo,
+  renderDryRunUpload,
+  renderFailedUpload,
+  renderFailedUploadBecauseOf403,
+  renderMissingEnvironmentVariable,
+  renderMissingParameter,
+  renderMissingReleaseVersionParameter,
+  renderSuccessfulCommand,
+  renderUnsupportedParameterValue,
+  renderUpload,
+} from './renderer'
+
+export class UploadCommand extends Command {
+  public static SUPPORTED_SOURCES = ['snyk']
+
+  public static usage = Command.Usage({
+    description: 'Upload dependencies graph to Datadog.',
+    details:
+      'Uploads dependencies graph to Datadog to detect runtime vulnerabilities by Continuous Profiler. See README for details.',
+    examples: [
+      [
+        'Upload dependency graph generated by `snyk test --print-deps --sub-project=my-project --json > ./snyk_deps.json` command',
+        'datadog-ci dependencies upload ./snyk_deps.json --source snyk --service my-service --release-version 1.234',
+      ],
+    ],
+  })
+
+  private static INVALID_INPUT_EXIT_CODE = 1
+  private static MISSING_FILE_EXIT_CODE = 2
+  private static UPLOAD_ERROR_EXIT_CODE = 3
+
+  private config = {
+    apiHost: process.env.DATADOG_API_HOST || 'api.datadoghq.com',
```

Comment:
Other commands uses `DATADOG_SITE` environment variable to select the correct datadog instance, could you use this instead?
Also, we should maybe have a `getApiUrlFromSite` function in helpers that all commands could use to obtain the correct api URL.

Reply:
I decided to use `DATADOG_API_HOST` for simplicity. There are some cons regarding the `DATADOG_SITE` variable:
 * we can't use `app.datad0g.com` host to test commands on staging (because with `DATADOG_SITE` we have to hardcode `api.` subdomain for prod)
 * we still have to set `DATADOG_API_HOST` to send metrics to the EU. I think it's a bad UX because if someone sends a dependency graph to the EU, he will also want to send metrics to the EU.

Here is a comparison between two README's that illustrates this:
## `sourcemap upload`:
It is possible to configure the tool to use Datadog EU by defining the `DATADOG_SITE` environment variable to datadoghq.eu. By defaut the requests are sent to Datadog US. In order to send the usage metrics to the correct datacenter, the `DATADOG_API_HOST` must also be set if another instance of Datadog is used. For example, for Datadog EU, it should be set to api.datadoghq.eu.

## `dependencies upload`:
By default, the command sends requests to Datadog US. To use Datadog EU, set `DATADOG_API_HOST` environment variable to `api.datadoghq.eu`.

Do you think I should use `DATADOG_SITE` for consistency, even if it's harder to set-up and explain? :)

Reply:
Ah, I read your comment again and now I see what you suggest - it makes sense :)